### DISCUSSION TITLE
Docs: depName vs. packageName in regex manager

### DISCUSSION THREAD

**Discussion by chludwig-haufe** [renovatebot/renovate#17604] [Suggest an Idea]
One distinguishing feature of the `regex` manager is that one can configure `depName` and `packageName` separately. All the [`regex` manager documentation](https://docs.renovatebot.com/modules/manager/regex/) says about this is the following, though: 
> You can optionally have a packageName capture group or a packageNameTemplate if it differs from depName

It does not say where these fields are used, respectively.

The rest of the documentation seems to use these terms interchangeably. In particular, the package rule configuration docs refer to _package names_ and do not mention `depName` - the [`PackageNameMatcher`](https://github.com/renovatebot/renovate/blob/4ccd085aa56fec3abcc3155eb80b942501dcff05/lib/util/package-rules/package-names.ts#L7), [`PackagePatternsMatcher`](https://github.com/renovatebot/renovate/blob/4ccd085aa56fec3abcc3155eb80b942501dcff05/lib/util/package-rules/package-patterns.ts#L10), and [`PackagePrefixesMatcher`](https://github.com/renovatebot/renovate/blob/4ccd085aa56fec3abcc3155eb80b942501dcff05/lib/util/package-rules/package-prefixes.ts#L7) do use the value of `depName`, though.

It would be helpful if the documentation stated the conceptual difference between `depName` and `packageName`. At the very least, the `regex` documentation could point out that the package rules are evaluated against `depName`. I am not the first to stumble over this, cf. #15654.

**Comment by HonkingGoose** [comment]
There's a PR to improve the regex manager docs here:

- https://github.com/renovatebot/renovate/pull/15067

I don't know if that PR fixes your problem though. 🙈

In any case, I agree that we should be clear about the difference between `depName` and `packageName`.

**Reply by chludwig-haufe** [reply]
For the most part, the PR #15067 you mentioned adds internal links – that's generally helpful. The PR does not clarify the use of `depName` vs. `packageName`, though.

For me, the confusing part was that the [`matchPackageName` package rule configuration](https://docs.renovatebot.com/configuration-options/#matchpackagenames) ignored the `packageName` I assigned via the `regex` manager. I don't mind that `matchPackageName` in fact matches against `depName` (this even serves my use case better), but it is for sure not self-explanatory.

I'd suggest to add a short section to the `regex` documentation along the lines of the following:
> ### `depName` vs. `packageName`
> 
> By default, the fields `depName` and `packageName` have the same value. You may override the value of `packageName` by means of either a `packageName` capture group or a `packageNameTemplate`.
>
> `packageName` is used to look up a dependency in the respective datasource. For instance, you use the package name `python/cpython` in order to refer to the CPython releases in the `github-tags` datasource or `my.private.registry/repo/image-name` to refer to an image `repo/image-name` in the registry `my.private.registry` using the `docker` datasource.
>
> `depName`, on the other hand, is the name assigned to a dependency. It is used, e.g., to create the [default branch topic](https://docs.renovatebot.com/configuration-options/#branchtopic). Despite their name, the package rule config options [`matchPackageNames`](https://docs.renovatebot.com/configuration-options/#matchpackagenames), [`matchPackagePatterns`](https://docs.renovatebot.com/configuration-options/#matchpackagepatterns), and [`matchPackagePrefixes`](https://docs.renovatebot.com/configuration-options/#matchpackageprefixes) match against the value of `depName`, not `packageName`.

**Reply by HonkingGoose** [reply]
@chludwig-haufe Can you copy/paste your idea into the PR I linked above, as a review comment? That way we don't lose your idea.

**Reply by viceice** [reply]
@HonkingGoose we working on switching those two names. so packageName becomes the primary filter and depName is only a visual short name

**Reply by HonkingGoose** [reply]
Oh okay!

**Reply by viceice** [reply]
- #16012
